src/defaults.mligo

```diff
@@ -2,42 +2,66 @@
 // SPDX-License-Identifier: LicenseRef-MIT-TQ
 
 #include "types.mligo"
+#include "proposal.mligo"
 
-let default_config : config = {
-    proposal_check = (fun (params, extras : propose_params * contract_extra) -> true);
-    rejected_proposal_return_value = (fun (proposal, extras : proposal * contract_extra) -> 0n);
-    decision_lambda = (fun (proposal, extras : proposal * contract_extra) -> (([] : (operation list)), extras));
-
-    max_proposals = 500n;
-    max_votes = 1000n;
-    max_quorum_threshold = {numerator = 99n; denominator = 100n}; // 99%
-    min_quorum_threshold = {numerator = 1n; denominator = 100n}; // 1%
-    max_voting_period = 2592000n; // 60 * 60 * 24 * 30
-    min_voting_period = 1n;
-    custom_entrypoints = (Map.empty : custom_entrypoints);
-    }
-
-let default_storage (admin , governance_token, now_val, metadata : address * governance_token * timestamp * metadata_map) : storage = {
-    ledger = (Big_map.empty : ledger);
-    operators = (Big_map.empty : operators);
-    governance_token = governance_token;
-    admin = admin;
-    pending_owner = admin;
-    metadata = metadata;
-    voting_period = 11n;
-    quorum_threshold = {numerator = 1n; denominator = 10n}; // 10%
-    extra = (Big_map.empty : (string, bytes) big_map);
-    proposals = (Big_map.empty : (proposal_key, proposal) big_map);
-    proposal_key_list_sort_by_date = (Set.empty : (timestamp * proposal_key) set);
-    permits_counter = 0n;
-    freeze_history = (Big_map.empty : freeze_history);
-    total_supply = Map.literal
-        [ (frozen_token_id, 0n)
-        ];
-    fixed_proposal_fee_in_token = 0n;
-    frozen_token_id = frozen_token_id;
-    last_period_change = {changed_on = now_val; period_num = 0n}
+let default_config (data : initial_config_data) : config = {
+  proposal_check = (fun (params, extras : propose_params * contract_extra) -> true);
+  rejected_proposal_return_value = (fun (proposal, extras : proposal * contract_extra) -> 0n);
+  decision_lambda = (fun (proposal, extras : proposal * contract_extra) -> (([] : (operation list)), extras));
+
+  max_proposals = 500n;
+  max_votes = 1000n;
+  max_quorum_threshold = data.max_quorum;
+  min_quorum_threshold = data.min_quorum;
+  max_voting_period = data.max_period;
+  min_voting_period = data.min_period;
+
+  custom_entrypoints = (Map.empty : custom_entrypoints);
+}
+
+let max_vp (a, b : voting_period * voting_period) : voting_period =
+  if a > b
+  then a
+  else b
+
+let min_vp (a, b : voting_period * voting_period) : voting_period =
+  if a < b
+  then a
+  else b
```

Comment:
Given that this is only used to bound the `voting_period`, I'd replace it with a single:
```suggestion
let bound_vp (vp, min_vp, Max_vp : voting_period * voting_period * voting_period)
    : voting_period =
  if vp < min_vp
  then min_vp
  else
    if vp > max_vp
    then max_vp
    else vp 
```